### ISSUE TITLE
Massive memory leak

### ISSUE THREAD

**Issue by RexfelisLOC** [AdsJumbo/AdsJumbo.com#28]
There's a massive memory leak which causes memory usage to go over 2gb within 10 mins of using the SDK.

**Comment by RexfelisLOC** [comment]
This needs to be resolved asap.  As it stands the SDK is no longer usable.  Within 15 mins approximately 2gb of memory is used up and apps start to crash on Xbox One.

**Comment by atecey** [comment]
It won't be fixed. I raised a couple of simple issues around documentation and they just closed them out after asking support if the project is dead or not. I was informed they would fix the issues in approx 30 days or so. They just closed out all the GitHub issues without actually fixing them. To date still no fixes....

If you ever find another ad provider that pays out a decent amount let please me know.

**Comment by tecnobillo** [comment]
Last December I had to abandon adsjumbo because my application was consuming 4gb of ram (8mb without adsjumbo...). I reported the bug and never got a response. I thought it was an IronPython related problem... but I can see that it is not. Now I am testing webview2 with google adsense.

**Comment by atecey** [comment]
@tecnobillo did you have any luck with webview2 and adsense? Do you get decent payouts? I tried Vungle a while back and while I got plenty of hits and clicks, I think I made like 2 cents for 1 week.

**Comment by tecnobillo** [comment]
It really doesn't work very well. Last July the page gave 0.03 US $ / 1000
sessions (I earned 0.70 $ in the whole month). At first I was earning a
little more, but as advertisers observe that no one clicks on that page, I
suppose they bid less and less. On the other hand, I am not clear if I am
breaking any adsense rule that prevents me from receiving the payment (if
one day I reach the minimum lol).

**Comment by AscendantWare** [comment]
Hi tecnobillo,

as far as i know, forbids google to show adsense ads on a desktop application. 
I am not aware of any other provider who provides advertising for desktop applications.
I wish i would found one. :/

**Comment by mmssem** [comment]
I also had the same problem with a C# winform program.
my program’s RAM usage is about 60M without advertisements, but after 10 minutes of using advertisements, it exceeds 1G and goes up endlessly. So I tried to ditch AdsJumbo.

But, I solved it like this. Now, no matter how long it is turned on, it does not exceed 160M.
(By the way, I'm not a programming expert. I'm just a hobbyist, so I don't know much about professional things.)

1. Variable declaration
AdsJumboWinForm.BannerAds bannerAds1;

2. Dynamically create bannerAds1 in Form1_Load
bannerAds1 = new AdsJumboWinForm.BannerAds();
this.Controls.Add(bannerAds1);
bannerAds1.Location = new System.Drawing.Point(7, 419);
bannerAds1.BringToFront();
bannerAds1.ShowAd(468, 60, "your Application ID");
timer1.Interval = 55000;
timer1.Enabled = true;

3. Add timer1
bannerAds1.Dispose();
GC.SuppressFinalize(this);
System.GC.Collect(0, GCCollectionMode.Forced);
System.GC.WaitForFullGCComplete();
timer2.Interval = 5000;
timer2.Enabled = true;

4. Add timer2
bannerAds1 = new AdsJumboWinForm.BannerAds();
this.Controls.Add(bannerAds1);
bannerAds1.ShowAd(468, 60, "your Application ID");
bannerAds1.Location = new System.Drawing.Point(7, 419);
bannerAds1.BringToFront();
timer2.Enabled = false;

Capture images and example files are at the link below.
https://mungkhs.tistory.com/116
